Main/src/main/java/com/main/acad/dao/ChapterDao.java

```diff
@@ -58,75 +78,88 @@ public void updateChapter(Chapter chapter) {
+            Statement statement = connection.createStatement();
+            ResultSet resultSet = statement.executeQuery(sqlGetAllQuery);
             while (resultSet.next()) {
-                chapter.setId(resultSet.getInt(1));
-                chapter.setName(resultSet.getString(2));
+                Chapter chapter = new Chapter();
+                chapter.setId(resultSet.getInt("id_chapter"));
+                chapter.setName(resultSet.getString("name"));
+                chaptersList.add(chapter);
             }
-            logger.info("User successfully get. User details: " + chapter);
+            connectionPool.surrenderConnection(connection);
+            logger.info("All Chapters successfully get. List chapters details : " + chaptersList.size());
         } catch (SQLException e) {
-            logger.info("connection have some error");
+            logger.info("An error occurred in the ChapterDao class in the getlistChapters method");
+        } catch (InterruptedException e) {
+            e.printStackTrace();
         }
-        throw new UnsupportedOperationException();
+        return chaptersList;
     }
 
     @Override
-    public List<Chapter> listChapters() {
-        List<Chapter> chaptersList = new ArrayList();
-        try (Connection connection = ConnectionPool.getDataSource().getConnection();
-             Statement statement = connection.createStatement();
-             ResultSet resultSet = statement.executeQuery(sqlGetAll);) {
+    public FileReader getInformstioAboutChildren(String name) {
+        try {
+            Connection connection = connectionPool.borrowConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(sqlGetByidQuery);
+            preparedStatement.setString(1, name);
+            ResultSet resultSet = preparedStatement.executeQuery();
             while (resultSet.next()) {
-                Chapter chapter = new Chapter();
-                chapter.setId(resultSet.getInt("id_chapter"));
-                chapter.setName(resultSet.getString("name"));
-                chaptersList.add(chapter);
+                return new FileReader(resultSet.getString("name"));
             }
-            logger.info("All Chapters successfully get. Chapters details: " + chaptersList.size());
-
-        } catch (SQLException e) {
-            logger.info("connection have some error");
+            connectionPool.surrenderConnection(connection);
+            logger.info("All information about chapter child successfully get.");
+        } catch (SQLException | IOException e) {
+            logger.info("An error occurred in the ChapterDao class in the getInformstioAboutChildren method");
+        } catch (InterruptedException e) {
+            e.printStackTrace();
         }
-        return chaptersList;
+        throw new ChapterDaoMethodhasError("An error occurred in the ChapterDao class in the getInformstioAboutChildren method");
```

Comment:
Common naming for exceptions: first business part, second - word "Exception", examples: NotWaterException, TooHotException, RainStoppedException. Naming should contain specific business meaning